### PR TITLE
Feature/dynamic keys

### PR DESCRIPTION
**Pull request by jluftcimpress**
dynamically pulling in id/secret key values from env variables. Defaulting to id/secret

### CODE REVIEW

**Comment by chrisoverzero** [comment]
Hi, @jluftcimpress. Can I ask, for what reason do you want to be able to change the names of these keys?

**Comment by jluftcimpress** [comment]
Hi Chris,

We (Alchemy) have unfortunately used "Auth0ClientID" and "Auth0ClientSecret" in all of our applications. The current version of the rotator expects the secrets to be stored as "id" and "secret". Rather than updating all of our secrets and services - we decided to adjust the lambda to dynamically inject these values to be both be backward compatible, and flexible for any format of OAuth secret storage in AWS. 

I actually had this change approved / merged on a forked branch of this repo (used by Vista), but we figure it is best to have it here as this is the source of truth we will be relying on moving forward. It should be a safe change that helps us a ton

**Comment by chrisoverzero** [comment]
I admit that I'm kind of reluctant to do this, since I don't know of any other rotation Lambda Function which does the same. But I want to give this a fair shake, so I'll wait to make a decision until after review and addressing.

File: rotator/rotator.py [hunk 1]
```diff
@@ -16,8 +16,10 @@
 EXCLUDE_CHARACTERS = r'''"%!'()*,/:;?@[\]`{|}~<>^&#$'''
 
 CLIENT_REGISTRY_TEMPLATE = URITemplate('https://clients.oauth.cimpress.io/v1/clients/{client_id}/secrets')
+ID_KEY = os.environ.get('CLIENT_ID_KEY') if os.environ.get('CLIENT_ID_KEY') is not None else "id"
```
Comment (chrisoverzero) [inline rotator/rotator.py]: Here and below should probably use the 2-arg version of `get`. Also, house style is single-quoted strings.

File: template.yml [hunk 2]
```diff
@@ -97,6 +97,8 @@ Resources:
           ISSUER: !Ref Issuer
           OVERLAP_DURATION: !Ref OverlapDuration
           SECRETS_MANAGER_ENDPOINT: !Ref Endpoint
+          CLIENT_SECRET_KEY: secret
```
Comment (chrisoverzero) [inline template.yml]: I think that these environment variable values are unconfigurable without a template parameter, right?

**Review by chrisoverzero — CHANGES_REQUESTED** [review]

File: rotator/rotator.py (hunk 1 above)
Comment (jluftcimpress) [reply rotator/rotator.py]: Great call, I'm rusty with my python - forgot about get with the default value. Pushed out the changes

File: template.yml (hunk 2 above)
Comment (jluftcimpress) [reply template.yml]: In the forked version of this repo - they are using this section to set env variables values without parameters.  

I added new parameters and the !Ref like the others are set up here. Should be better now

**Comment by jluftcimpress** [comment]
> I admit that I'm kind of reluctant to do this, since I don't know of any other rotation Lambda Function which does the same. But I want to give this a fair shake, so I'll wait to make a decision until after review and addressing.

I made all of the changes based on your feedback. I do think this change could be useful for people who want to store their id and secret with different names like we currently do.

**Comment by chrisoverzero** [comment]
I think I do approve of this. I'd like to change up some wording and add the ephemera for release, but I'm not sure if I… If _my_ commit is signed, but yours isn't, will it still merge? Any chance you can sign your commits so far? Then I'll do the administrivia.

**Review by chrisoverzero — APPROVED** [review]

**Comment by jluftcimpress** [comment]
Hey Chris - Should be all set now. Sorry I got caught up in a few other tickets.

Edit: I see I need to sign all of the previous commits as well, I will work on that shortly

**Pull request closed without merging**